quickjs.c

```diff
@@ -40007,10 +40021,44 @@ static JSValue js_iterator_from(JSContext *ctx, JSValue this_val,
     return iter;
 }
 
+static JSValue js_create_iterator_helper(JSContext *ctx, JSValue iterator,
+                                         JSIteratorHelperKindEnum kind, JSValue func, int64_t limit);
+
 static JSValue js_iterator_proto_drop(JSContext *ctx, JSValue this_val,
                                       int argc, JSValue *argv)
 {
-    return JS_ThrowInternalError(ctx, "TODO implement Iterator.prototype.drop");
+    JSValue v;
+    double dlimit;
+    int64_t limit;
+    if (!JS_IsObject(this_val))
+        return JS_ThrowTypeError(ctx, "Iterator.prototype.drop called on non-object");
+    v = JS_ToNumber(ctx, argv[0]);
+    if (JS_IsException(v))
+        return JS_EXCEPTION;
+    if (JS_VALUE_IS_NAN(v))
+        goto fail;
+    // Check for Infinity.
+    if (JS_ToFloat64(ctx, &dlimit, v)) {
+        JS_FreeValue(ctx, v);
+        return JS_EXCEPTION;
+    }
+    if (!isfinite(dlimit)) {
+        JS_FreeValue(ctx, v);
+        if (dlimit < 0)
+            goto fail;
+        else
+            limit = MAX_SAFE_INTEGER;
+    } else {
+        v = JS_ToIntegerFree(ctx, v);
+        if (JS_IsException(v))
+            return JS_EXCEPTION;
+        if (JS_ToInt64Free(ctx, &limit, v))
+            return JS_EXCEPTION;
+    }
+    if (limit < 0)
+fail:
+        return JS_ThrowRangeError(ctx, "Must be positive");
```

Comment:
For consistency:
```suggestion
    if (limit < 0) {
    fail:
        return JS_ThrowRangeError(ctx, "must be positive");
    }
```
Also s/Must/must/ because I don't think we capitalize error messages.